src/functions.php

```diff
@@ -15,13 +15,13 @@
  *
  * @param callable(mixed ...$args):mixed $function
  * @param mixed ...$args Optional list of additional arguments that will be passed to the given `$function` as is
- * @return PromiseInterface<mixed>
+ * @return callable(): PromiseInterface<mixed>
  * @since 4.0.0
  * @see coroutine()
  */
-function async(callable $function, mixed ...$args): PromiseInterface
+function async(callable $function): callable
 {
-    return new Promise(function (callable $resolve, callable $reject) use ($function, $args): void {
+    return static fn (mixed ...$args): PromiseInterface => new Promise(function (callable $resolve, callable $reject) use ($function, $args): void {
```

Comment:
```suggestion
    return static fn (): PromiseInterface => new Promise(function (callable $resolve, callable $reject) use ($function): void {
```

as discussed here: https://github.com/reactphp/async/pull/19/files#r773363277

Reply:
@azjezz Only dropped the docblock item as this suggested change will break it

Reply:
@azjezz The changes LGTM as is, there's definitely value in being able to pass additional arguments through to the original function.